addon/validators/base.js

```diff
@@ -56,17 +55,32 @@ export default Ember.Object.extend({
       }
     });
   },
+
+  handlePropertyRetrievalError(error) {
+    this.errors.pushObject(error);
+  },
+
   _validate: Ember.on('init', function() {
     this.errors.clear();
-    if (this.canValidate()) {
-      this.call();
-    }
-    if (get(this, 'isValid')) {
-      return Ember.RSVP.resolve(true);
-    } else {
-      return Ember.RSVP.resolve(false);
+    const isValid = get(this, 'isValid');
+
+    if (!this.canValidate()) {
+      return Ember.RSVP.resolve(isValid);
     }
+
+    const valuePromise = get(this.model, this.property);
+    return new Ember.RSVP.Promise((resolve) => {
+      resolve(valuePromise);
+    }).then((results) => {
+      this.errors.clear();
+      this.call(results);
```

Comment:
The way this works is each of the individual validators, like `absence` and `numericality`, have a `call` method to invoke them.

This `base` validator goes through all the different validations that are specified in that object, ie:

```
'mostRecentWeightForUser': {
  presence: true,
  numericality: { greaterThan: 0 }
}
```

and calls `validate` on them, which calls `_validate`, which in turn calls `call`.

I'm not totally sure why they separated `validate` from `_validate`. Maybe the ability to allow people the base `validate` method while keeping all the core "validation" functionality.